timdex_dataset_api/utils.py

```diff
@@ -0,0 +1,76 @@
+"""timdex_dataset_api/utils.py"""
+
+import logging
+import os
+import pathlib
+from urllib.parse import urlparse
+
+import boto3
+from mypy_boto3_s3.service_resource import S3ServiceResource
+
+logger = logging.getLogger(__name__)
+
+
+class S3Client:
+    def __init__(
+        self,
+    ) -> None:
+        self.resource = self._create_resource()
```

Comment:
Curious about the pivot from using `boto3`'s `Client` (what we typically use) vs. `Resource`?

Reply:
I opted for the `Resource` route here, as it just felt like a better fit?  But it may not always be!

Most specifically, for the `delete_folder` method:
```
bucket_obj = self.resource.Bucket(bucket)
receipt = bucket_obj.objects.filter(Prefix=prefix).delete()
``` 

Where I don't think the client approach has `.objects.filter()`?  It was convenient.  But _then_ it felt prudent to fully Resource in this utility class.